### ISSUE TITLE
ECPs missing from SAD guess

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#1473]
The atomic UHF solver code doesn't include ECP elements in its core Hamiltonian operator at present. Likewise, ECPs aren't taken into account in the spin state determination for the SAD state. (This is not a problem for #1458 though since it switches the default to spin-averaged densities.)

For instance,
```
molecule {
0 1
Xe
}

set guess sad
set basis def2-svp
set sad_print 2
energy('scf')
```

gives
```
  @Atomic UHF Final Energy for atom XE: -2507.62764642306365
Finished UHF Computation!

  ==> Iterations <==

                           Total Energy        Delta E     RMS |[F,P]|

   @DF-RHF iter   0:   285.32485664068048    2.85325e+02   0.00000e+00 
   @DF-RHF iter   1:  -189.84500624892323   -4.75170e+02   8.80223e-01 DIIS
   @DF-RHF iter   2:  -290.52105276490687   -1.00676e+02   4.66004e-01 DIIS
   @DF-RHF iter   3:  -328.09971971455047   -3.75787e+01   4.86655e-02 DIIS
   @DF-RHF iter   4:  -328.29180563637317   -1.92086e-01   6.06357e-03 DIIS
   @DF-RHF iter   5:  -328.29809735789945   -6.29172e-03   1.54650e-03 DIIS
   @DF-RHF iter   6:  -328.29834344170320   -2.46084e-04   2.46581e-04 DIIS
   @DF-RHF iter   7:  -328.29835685761799   -1.34159e-05   4.28492e-05 DIIS
   @DF-RHF iter   8:  -328.29835732243424   -4.64816e-07   4.48232e-06 DIIS
   @DF-RHF iter   9:  -328.29835733488505   -1.24508e-08   4.61031e-07 DIIS
  Energy converged.
```

so it's only a small 2200 Hartree difference between the calculations :tongue:

**Comment by susilehtola** [comment]
This should be a trivial fix, however.

**Comment by susilehtola** [comment]
@loriab located the problem at https://github.com/psi4/psi4/blob/master/psi4/driver/qcdb/libmintsbasisset.py#L946. The ECP basis sets aren't passed to the SAD code at all.

**Comment by susilehtola** [comment]
This was fixed in #1458 .
